src/Microsoft.TestPlatform.CommunicationUtilities/JsonDataSerializer.cs

```diff
@@ -152,6 +152,13 @@ public string SerializePayload(string messageType, object payload, int version)
             return JsonConvert.SerializeObject(message);
         }
 
+        /// <inheritdoc/>
```

Comment:
Please write a comment on why we chose this approach, just for reference.
Also consider making it an extension method.

Reply:
There are lot of discussion over stackoverflow, how to do deep copy.
- https://stackoverflow.com/questions/129389/how-do-you-do-a-deep-copy-of-an-object-in-net-c-specifically  
- https://stackoverflow.com/questions/78536/deep-cloning-objects

1. BinaryFormatter
This very generic and easy to implement, But it is not available in NS14.  
2. ICloneable
This required changes in all the class which are involved and difficult to maintain(complicated code).

As we are already using `JsonDataSeializer` to transfer the object over network. We can use the same functionality to do clone Which is well tested.